src/devices/GrovePi/GrovePi.cs

```diff
@@ -0,0 +1,190 @@
+﻿using Iot.Device.GrovePiDevice.Models;
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using System.Threading;
+
+namespace Iot.Device.GrovePiDevice
+{
+    /// <summary>
+    /// Create a GrovePi class
+    /// </summary>
+    public class GrovePi : IDisposable
+    {        
+        private I2cDevice _i2cDevice;
+        private readonly bool _autoDispose;
+
+        /// <summary>
+        /// The default GrovePi I2C address is 0x04
+        /// Other addresses can be use, see GrovePi documentation
+        /// </summary>
+        public static byte GrovePiSefaultI2cAddress => 0x04;
+
+        /// <summary>
+        /// The maximum ADC, 12 bit so 2013 on GrovePi
+        /// </summary>
+        public int MaxAdc => 1023;
+        /// <summary>
+        /// Contains the GrovePi key information
+        /// </summary>
+        public GrovePiInfo GrovePiInfo { get; internal set; }
+
+        /// <summary>
+        /// GrovePi constructor
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device. Device address is 0x04</param>
+        /// <param name="autoDispose">True to dispose the I2C device when disposing GrovePi</param>
+        public GrovePi(I2cDevice i2cDevice, bool autoDispose = true)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentException("I2C device can't be null");
+            _autoDispose = autoDispose;
+            GrovePiInfo = new GrovePiInfo() { SoftwareVersion = GetFirmwareVerion() };
+        }
+
+        public void Dispose()
+        {
+            if(_autoDispose)
+            {
+                _i2cDevice.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Get the firmware version
+        /// </summary>
+        /// <returns>GroovePi firmware version</returns>
+        public Version GetFirmwareVerion()
+        {
+            WriteCommand(GrovePiCommands.Version, 0, 0, 0);
+            Thread.Sleep(10);
+            var inArray = ReadCommand(GrovePiCommands.Version, 0);
+            return new Version(inArray[1], inArray[2], inArray[2]);
```

Comment:
`inArray[1], inArray[2], inArray[3]`?

Reply:
the first inArray is always 0. And it is not part of the version. Only the last 3 bytes are

Reply:
@Ellerbach What I meant is that you got `[2]` twice

Reply:
hooooooooooooooo